### PR TITLE
[test] added lazy load functionality

### PR DESCRIPTION
**Pull request by hamoly**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,92 +1,91 @@
...
+const App = () => {
+  const [link, setLink] = useState(process.env.REACT_APP_API_FETCH_URL);
+  const [error, setError] = useState(null);
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [movies, setMovies] = useState([]);
+  const [query, setQuery] = useState('');
+  const handleChange = (e) => {
+    if(e.target.value !== '') {
+      setQuery(e.target.value);
+      setLink(`${process.env.REACT_APP_API_SEARCH_URL}${e.target.value}`);
```
Comment (xfoxawy) [inline src/App.js:16]: move this logic to a function names `set` or `set_fetch_link` or whatever

File: src/App.js [hunk 2]
```diff
@@ -1,92 +1,91 @@
...
     }
-  }
-  getMovies(link)  {
+  };
+  const MovieAppRenderer = () => {
+    if (error) {
+      return (
+        <ShowMsg msg='ERROR : Please check you internet connection then reload the page' />
+      );
+    } else if (!isLoaded) {
```
Comment (xfoxawy) [inline src/App.js:27]: nitpick :  u dont need `else if` if the pervious condition does a return

File: src/App.js [hunk 3]
```diff
@@ -1,92 +1,91 @@
...
+      );
+    } else if (movies.length == 0) {
+      return (
+        <ShowMsg msg="Sorry couldn't find any search results ... please try another title …" />
+      );
+    } else {
+      return (
+        movies.map(movie => (
+            <MovieListApp key={movie.id} poster_path={movie.poster_path} title={movie.title} release_date={movie.release_date} overview={movie.overview} popularity={movie.popularity} vote_count={movie.vote_count} />
+            ))
```
Comment (xfoxawy) [inline src/App.js:39]: this can be moved to a separate component

File: src/movieapprenderer.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import ShowMsg from './components/showmsg';
+
+const MovieAppRenderer = () => {
+    if (error) {
+      return (
+        <ShowMsg msg='ERROR : Please check you internet connection then reload the page' />
+      );
+    } else if (!isLoaded) {
+      return (
+        <ShowMsg msg='Please wait while fetching movies ...' />
```
Comment (xfoxawy) [inline src/movieapprenderer.js:12]: this logic is repeated , can we `dry` it ?
search the word `dry code`

**Comment by AhmedTammam** [comment]
please fix conflicts

File: src/App.js [hunk 5]
```diff
@@ -1,92 +1,91 @@
...
-  constructor(props) {
-    super(props)
-    // state
-    this.state = {
-      error : null,
-      isLoaded : false,
-      movies : []
-      
+const App = () => {
+  const [link, setLink] = useState(process.env.REACT_APP_API_FETCH_URL);
```
Comment (AhmedTammam) [inline src/App.js:8]: why you use hooks, and it's not useful

**Pull request closed without merging**